From 42bf891c28bc54435c3eac37c4a0bc3bb85cee2c Mon Sep 17 00:00:00 2001
From: 456vv <123@456vv.com>
Date: Fri, 17 Jan 2025 11:43:05 +0800
Subject: [PATCH] all

---
 bindings/gumjs/generate-runtime.py | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/bindings/gumjs/generate-runtime.py b/bindings/gumjs/generate-runtime.py
index 30c643c..895fdd5 100644
--- a/bindings/gumjs/generate-runtime.py
+++ b/bindings/gumjs/generate-runtime.py
@@ -57,9 +57,17 @@ def generate_runtime(output_dir, priv_dir, input_dir, gum_dir, capstone_incdir,
                        cwd=priv_dir,
                        check=True)
         js_dir = priv_dir / "node_modules" / "frida-java-bridge" / "lib" 
+
         #(js_dir / "android.js").write_text("暂时不改这里", encoding="utf-8")
+
+        #在 Java.api.exports 使用更多内置方法
+        subprocess.run(f"sed -i 's/module.exports/module.exports = temporaryApi.exports/g' {(js_dir / 'android.js')}", shell=True)
+
+        #这条是可以使用 Interceptor.revert(addr) 来撤掉，所以不处理它
         #subprocess.run("sed -i '/Interceptor.attach(prettyMethod.impl, artController.hooks.ArtMethod.prettyMethod);/,/Interceptor.flush();/{d}' %s" % (js_dir / "android.js"), shell=True)
-        subprocess.run("sed -i 's/Current thread is not attached to the Java VM; please move this code inside a Java.perform() callback/cUrrent th^ead 1s n0t attAched t0 tHe ja{a v3; p1ease m0ve th1s c0de 1ns1de a JbAb.perf0rm[] ca11back/g ' %s" % (js_dir / "vm.js"), shell=True)
+
+        #这个用处不清楚
+        subprocess.run(f"sed -i 's/Current thread is not attached to the Java VM; please move this code inside a Java.perform() callback/VM Error/g ' {(js_dir / 'vm.js')}", shell=True)
 
     runtime_reldir = Path("runtime")
     runtime_srcdir = input_dir / runtime_reldir
-- 
2.47.0.windows.2

