From 4505107294d7e64441b67fee0a0961307ec37a2d Mon Sep 17 00:00:00 2001
From: 456vv <123@456vv.com>
Date: Thu, 31 Oct 2024 10:50:40 +0800
Subject: [PATCH] all

---
 gum/gum.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gum/gum.c b/gum/gum.c
index 239c70d..0af99c9 100644
--- a/gum/gum.c
+++ b/gum/gum.c
@@ -306,7 +306,7 @@ gum_init_embedded (void)
   g_log_set_default_handler (gum_on_log_message, NULL);
   gum_do_init ();
 
-  g_set_prgname ("frida");
+  g_set_prgname ("${FRIDA_PREFIX}");
 
 #if defined (HAVE_LINUX) && defined (HAVE_GLIBC)
   gum_libdl_prevent_unload ();
-- 
2.47.0.windows.2

